FastSimulation/CaloGeometryTools/src/CaloDirectionOperations.cc

```diff
@@ -1,13 +1,16 @@
+#include <cassert>
 #include "FastSimulation/CaloGeometryTools/interface/CaloDirectionOperations.h"
 
 CaloDirection CaloDirectionOperations::add2d(const CaloDirection& dir1, const CaloDirection& dir2) {
-  //  unsigned d1=Side(dir1);
-  //  unsigned d2=Side(dir2);
+  unsigned d1 = Side(dir1);
+  unsigned d2 = Side(dir2);
+  assert(d1 != 999);
+  assert(d2 != 999);
   constexpr CaloDirection tab[4][4] = {{NORTH, NORTHEAST, NONE, NORTHWEST},
                                        {NORTHEAST, EAST, SOUTHEAST, NONE},
                                        {NONE, SOUTHEAST, SOUTH, SOUTHWEST},
                                        {NORTHWEST, NONE, SOUTHWEST, WEST}};
-  return tab[Side(dir1)][Side(dir2)];
+  return tab[d1][d2];
```

Comment:
@cms-sw/fastsim-l2 Could you please review this function - what happens (if it can happen) if either dir1 or dir2 is `DOWN` (in which case d1 or d2 will be 5, outside of `tab` array bounds)?

Reply:
This code is from 2006: https://github.com/cms-sw/cmssw/commits/CMSSW_6_2_X/FastSimulation/CaloGeometryTools/src/CaloDirectionOperations.cc
It is currently only called in https://github.com/cms-sw/cmssw/blob/master/FastSimulation/CaloHitMakers/src/EcalHitMaker.cc.
We should indeed audit whether any out of bounds values are ever passed into this function. Please open an issue, as this will not happen on the timescale of merging this PR with multiple analyzer cleanups.

Reply:
@kpedro88 accessing array outside it's bounds is not guaranteed to crash the code (undefined behavior). Would it be OK to raise assertion if d1 or d2 point outside of array bounds?

Reply:
Given that we have not observed any crashes or obvious corruption so far related to this, I would prefer to relegate this topic to an issue, so that we can perform a thorough audit of all the values that could possibly come from EcalHitMaker, as noted above. Adding an assert could break workflows, which would then require immediate attention. We need to be able to plan and prioritize our development and maintenance time.

Reply:
@kpedro88 sure. I will revert the changes for now, just to avoid any possible crashes.